test/integration_test.go

```diff
@@ -0,0 +1,60 @@
+package test
+
+import (
+	"testing"
+
+	"github.com/google/cel-go/checker"
+	"github.com/google/cel-go/checker/decls"
+	"github.com/google/cel-go/common"
+	"github.com/google/cel-go/common/packages"
+	"github.com/google/cel-go/common/types"
+	"github.com/google/cel-go/interpreter"
+	"github.com/google/cel-go/parser"
+)
+
+func Test_stringOverfloats(t *testing.T) {
+	for _, expression := range []string{
+		"req_params.Nick.matches('k.*')",
+		"nick.matches('k.*')",
+		"nick in ['klmno', 'kpacha']",
+	} {
+		source := common.NewTextSource(expression)
+		p, errors := parser.Parse(source)
```

Comment:
There's a helper method in `interpreter_test.go` called `parseExpr()` which could be used to replace the parsing portion of this test case. However, since you're also checking, I would create a `compileExpr()` helper in `interpreter_test.go` which combines the parse and check steps:

```go
func compileExpr(t *testing.T, src string, decls ...*exprpb.Decl) *exprpb.CheckedExpr {
    t.Helper()
    s := common.NewTextSource(src)
    parsed, errors := parser.Parse(s)
    if len(errors.GetErrors()) != 0 {
        t.Error(errors.ToDisplayString())
        return nil
    }
    env := checker.NewStandardEnv(packages.DefaultPackage, types.NewProvider())
    env.Add(decls...)
    checked, errors := checker.Check(parsed, s, env)
    if len(errors.GetErrors()) != 0 {
       t.Error(errors.ToDisplayString())
       return nil
    }
    return checked
}
```

Then your test setup would probably looks as follows: 

```go
expr := compileExpr(t, "nick in ...", decls.NewIdent("nick", decls.String, nil))
prog := NewCheckedProgram(expr)
eval := interpreter.NewInterpretable(prog)
for _, tc := range []struct { } { ... }
```

Reply:
great! thx